imports/plugins/core/orders/server/no-meteor/queries/ordersByAccountId.js

```diff
@@ -0,0 +1,39 @@
+import ReactionError from "@reactioncommerce/reaction-error";
+
+/**
+ * @name ordersByAccountId
+ * @method
+ * @memberof Order/NoMeteorQueries
+ * @summary Query the Orders collection for orders made by the provided accountId and (optionally) shopIds
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} params - request parameters
+ * @param {String} params.accountId - Account ID to search orders for
+ * @param {String} params.shopIds - Shop IDs for the shops that owns the orders
+ * @return {Promise<Object>|undefined} - An Array of Order documents, if found
+ */
+export default async function ordersByAccountId(context, { accountId, shopIds } = {}) {
+  const { contextAccountId, collections, userHasPermission } = context;
+  const { Orders } = collections;
+
+
+  if (!accountId) {
+    throw new ReactionError("invalid-param", "You must provide accountId arguments");
+  }
+
+  // TODO: EK - Permissions check here
+  // Unless you are an admin with orders permission, you are limited to seeing it if you placed it
+  // if (!userHasPermission(["orders"], shopId)) {
```

Comment:
@aldeed is there a way to use multiple shopIds here for this permissions check?

Reply:
No, you'll have to do the `userHasPermission` check in a `shopIds.forEach` loop.

On line 15, it should be `accountId: contextAccountId`

I would not change `accountId` since that could produce unexpected results. You could just wrap the whole permissions check loop in `if (accountId !== contextAccountId)`, like this:

```js
if (accountId !== contextAccountId) {
  shopIds.forEach((shopId) => {
    if (!userHasPermission(["orders"], shopId)) {
      throw new ReactionError("access-denied", "Access Denied");
    }
  });
}
```

Reply:
In this case, if `accountId === contextAccountId`, will it not just skip the check then?

Reply:
Actually I don't think my example handles the case where they have "orders" permission and they request all orders for another account, without passing any `shopIds`. I can think of two possible solutions to that:
- Look up the orders, get all the shopIds on the orders, and then run permission check for each
- Set `shopIds` to the list of IDs for which they user has "orders" permission.

The first seems like a lot of work, so I like the second. You'd have to add a new permission function like `shopsUserHasPermissionFor` and it would become something like this:

```js
if (accountId !== contextAccountId) {
  // If an admin wants all orders for an account, we force it to be limited to the
  // shops for which they're allowed to see orders.
  if (!shopIds) shopIds = shopsUserHasPermissionFor("orders");

  shopIds.forEach((shopId) => {
    if (!userHasPermission(["orders"], shopId)) {
      throw new ReactionError("access-denied", "Access Denied");
    }
  });
}
```

Reply:
@kieckhafer If `accountId === contextAccountId` why would any checks be necessary? You should be able to see all of your own orders regardless of what permissions you have.